File: solutions/typescript/beer-song/1/beer-song.ts
export function verse(index: number): string {
  let verse = '';
  switch (index) {
    case 0:
      verse = 'No more bottles of beer on the wall, no more bottles of beer.\n' +
        'Go to the store and buy some more, 99 bottles of beer on the wall.\n';
      break;
    case 1:
      verse = '1 bottle of beer on the wall, 1 bottle of beer.\n' +
        'Take it down and pass it around, no more bottles of beer on the wall.\n';
      break;
    case 2:
      verse = '2 bottles of beer on the wall, 2 bottles of beer.\n' +
        'Take one down and pass it around, 1 bottle of beer on the wall.\n';
      break;
    default:
      verse = `${index} bottles of beer on the wall, ${index} bottles of beer.\n` +
        `Take one down and pass it around, ${index - 1} bottles of beer on the wall.\n`;
      break;
  }

  return verse;
  
}

export function sing(
  initialBottlesCount?: number,
  takeDownCount?:number 
): string {
  let song = '';
  let bottlesCount = initialBottlesCount || 99;
  let takeDown = takeDownCount || 0;
  for (let i = bottlesCount; i >= takeDown; i -= 1) {
    song += verse(i);
    if (i !== takeDown) {
      console.log("I am here")
      song += '\n';
    }
  }
  console.log(song);

  return song;
}

File: solutions/c/beer-song/1/beer_song.c
#include "beer_song.h"
#include <stdio.h>


void recite(uint8_t start_bottles, uint8_t take_down, char **song) {
    int slot = 0;
    for (uint8_t i = 0; i < take_down; i++) {
        uint8_t bottles = start_bottles - i;
        if (bottles > 1) {
            sprintf(song[slot], "%d bottles of beer on the wall, %d bottles of beer.", bottles, bottles);
            sprintf(song[++slot], "Take one down and pass it around, %d %s of beer on the wall.", bottles - 1, (bottles - 1 == 1) ? "bottle" : "bottles");
            sprintf(song[++slot], "");
        } else if (bottles == 1) {
            sprintf(song[slot], "1 bottle of beer on the wall, 1 bottle of beer.");
            sprintf(song[++slot], "Take it down and pass it around, no more bottles of beer on the wall.");
            sprintf(song[++slot], "");
        } else {
            sprintf(song[slot], "No more bottles of beer on the wall, no more bottles of beer.");
            sprintf(song[++slot], "Go to the store and buy some more, 99 bottles of beer on the wall.");
        }
        slot++;
    }
}
